### ISSUE TITLE
Review AI non-handicap based cheats 

### ISSUE THREAD

**Issue by ErikSandbraaten** [Religion-And-Revolution-Extended/Mod#88]
This is intended as a general list of AI cheats that do depend on the difficulty. In other words, the AI receives the full benefit of these even when playing on explorer. Keep in mind that this list is probably not complete.
It is unlikely that we'll ever get rid of all of these, but it should be a goal to reduce them!

1) Slaves and prisoners can found cities CvUnit::canFound
2) Can breed livestock despite having none  CvCity::setYieldStored / CvPlot::calculatePotentialYield
3) No building or unit production decay doDecay()
4) Automagic distribution of certain yields redistributeWood, redeistributeMuskets /( and cannons)
5) Can change cannons to muskets and muskets to cannnons
( from CvPlayer::doTurn)
6) Can build wagons with only hammersCvPlayer::getYieldProductionNeeded
7) No negative effects from negative city health  CvCity::getCityHealthChangeFromPopulation, void CvCity::setCityHealth(int iValue)
8) Free production (overflow) when founding a city CvCity::init
9) Less time required to become expert CvCity::LbD_try_become_expert
10) Timer is not reset when changing profession (see above)
11) Probably necessary since the AI job assigner will shuffle units “around”
12) Criminals and similar units cannot escape CvCity::LbD_try_escape
13) Reduced REF option is always active (REF reduced by 50%)  CvPlayer::NBMOD_REF_MakeStartValue
14) No unprotected city unrest and half price for militia CvPlayer::checkForMilitiaOrUnrest
15) Discount for buying native mercenaries  CvPlayer::checkForNativeMercs
16) Discount for buying mercenaries (conquistadors)  CvPlayer::checkForConquistadors
17) Less pirates spawn near AI cities CvPlayer::checkForPirates
18) Free military unit if fighting a church war CvPlayer::checkForChurchWar
19) Possible free unit when founding a city CvUnit::doFound
20) Natives get free gold per turn  CvPlayer::doTurn
21 Free hammer overflow when AI players creates city.
void CvCity::init(int iID, PlayerTypes eOwner, int iX, int iY, bool bBumpUnits)
Controlled by changeOverflowProduction(GC.getDefineINT("INITIAL_AI_CITY_PRODUCTION"), 0);

**Comment by ErikSandbraaten** [comment]
After posting this list I am happy to report that we actually removed cheat 1) recently without impacting the AI performance. We still need to verify it properly though since there are many AI "loopholes".

**Comment by abmpicoli** [comment]
Notice that these cheats are places just because the AI sucks... as always... Maybe we should keep them? Unless you build an AI based on HAL9000 :)

**Comment by ErikSandbraaten** [comment]
Well, we're forced to keeping (most) of them for a while until the AI improved sufficiently. Note that some of them are a bit outrageous like automatic distribution and transmution of yield. It's almost like the AI is playing a different game since it is allowed to straight up ignore certain game mechanics. 
I think that, without doubt, that the colonization AI has the most cheats of any civ game. In civ4..6 the AI generally only received boost to production, research etc rather than being allowed to ignore fundamental game rules.

**Comment by abmpicoli** [comment]
Yep. Notice that colonization is a much more complex game, with all the logistics involved... I have the feel that, if we are going to make a really efficient AI, we will have to multithread this game somehow.
